agent/internal/config/jwt.go

```diff
@@ -0,0 +1,77 @@
+package config
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+type JWTToken struct {
+	Header        JWTHeader
+	Payload       JWTPayload
+	Signature     string
+	InitialString string
+}
+
+type JWTHeader struct {
+	Alg string `json:"alg"`
+	Typ string `json:"typ,omitempty"`
+}
+
+type JWTPayload struct {
+	Iss string `json:"iss,omitempty"`
+	Sub string `json:"sub,omitempty"`
+	Aud string `json:"aud,omitempty"`
+	Exp string `json:"exp,omitempty"`
+	Iat int    `json:"iat,omitempty"`
+}
+
+func (field *JWTToken) SetValue(unvalidatedToken string) error {
+	if unvalidatedToken == "" {
+		return fmt.Errorf("grpc jwt token can not be unset")
+	}
+
+	token, err := ValidateAndCreateJWT(unvalidatedToken)
+	if err != nil {
+		return err
+	}
+
+	*field = token
+	return nil
+}
+
+func ValidateAndCreateJWT(unvalidatedToken string) (JWTToken, error) {
+	splitted := strings.Split(unvalidatedToken, ".")
```

Comment:
`splitted` is not a correct English word, use a correct one instead.

Reply:
[changed to `split` ](https://preply.com/en/question/past-tense-of-split)